docker/errors.go

```diff
@@ -17,6 +17,14 @@ var (
 	ErrTooManyRequests = errors.New("too many requests to registry")
 )
 
+// ErrBadRequest is returned when the status code returned is 400
+type ErrBadRequest struct {
```

Comment:
Is there any user for this? If this is supposed to be a part of the `GetBlobAt` interface, the caller is probably not going to be checking for a docker-transport-specific error.

Reply:
`storage/chunked_zstd.go` is using it

Reply:
Oops, GitHub is auto-collapsing files…

I don’t think it makes sense to have a generic interface with a specific error condition like that; compare `types.ManifestTypeRejectedError`, as an explicitly documented error case in the `PutManifest` method.

(Alternatively, it would be consistent to just say “this is totally docker+c/storage-specific, there will be no generic interfaces, and this is a `DockerGetBlobAt`, and so on. That’s probably not the best design, I expect `dir:` could easily implement `GetBlobAt`, for example.)